mix/CodeModel.cpp

```diff
@@ -314,6 +320,49 @@ void CodeModel::runCompilationJob(int _jobId)
 	emit stateChanged();
 }
 
+void CodeModel::gasEstimation(solidity::CompilerStack const& _cs)
+{
+	m_gasCostsMaps.clear();
+	for (std::string n: _cs.getContractNames())
+	{
+
+		ContractDefinition const& contractDefinition = _cs.getContractDefinition(n);
+		QString sourceName = QString::fromStdString(*contractDefinition.getLocation().sourceName);
+
+		if (!m_gasCostsMaps.contains(sourceName))
+			m_gasCostsMaps.insert(sourceName, QVariantList());
+
+		if (!contractDefinition.isFullyImplemented())
+			continue;
+		dev::solidity::SourceUnit const& sourceUnit = _cs.getAST(*contractDefinition.getLocation().sourceName);
+		AssemblyItems const* items = _cs.getRuntimeAssemblyItems(n);
+
+		StructuralGasEstimator estimator;
+		std::map<ASTNode const*, GasMeter::GasConsumption> gasCosts = estimator.breakToStatementLevel(estimator.performEstimation(*items, std::vector<ASTNode const*>({&sourceUnit})), {&sourceUnit});
+
+		for (auto gasIte = gasCosts.begin(); gasIte != gasCosts.end(); ++gasIte)
+		{
+			SourceLocation location = gasIte->first->getLocation();
+			GasMeter::GasConsumption cost = gasIte->second;
+			GasMap* gas = new GasMap(location.start, location.end, (new QBigInt(cost.value))->value());
```

Comment:
`cost` can be infinite here. Perhaps it is better to already store only categories in the map?